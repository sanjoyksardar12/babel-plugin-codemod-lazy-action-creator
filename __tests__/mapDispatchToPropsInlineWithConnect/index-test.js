const babel = require("@babel/core");
const path = require("path");
const plugin = require("../../index.js");

// import moduleName, {foo} from '../moduleName';

// jest.mock("../../src/actions/multiplication.js", () => {
//   return {
//     __esModule: true,
//     default: jest.fn(() => "Default multiplication of 2 nos."),
//     foo: jest.fn(() => "multiplication of 3 nos."),
//   };
// });

// moduleName(); // Will return 42
// foo();

const babelConfig = { presets: [ "@babel/preset-react"], plugins: [plugin] };

// jest.mock("./actions/multiplication.js",()=>({
//   default: ()=>"Default multiplication of 2 nos.",
//   multiplication3: ()=> "multiplication of 3 nos."
// }));

it("MapDispatchToProps as inline with connnect", () => {
  const { code } = babel.transformFileSync(path.resolve(__dirname, "../../src/inlineWithConnect.js"), babelConfig);
  expect(code).toMatchSnapshot();
});

it("MapDispatchToProps as object", () => {
  const { code } = babel.transformFileSync(path.resolve(__dirname, "../../src/asObject.js"), babelConfig);
  expect(code).toMatchSnapshot();
});

it("MapDispatchToProps as fuction", () => {
  const { code } = babel.transformFileSync(path.resolve(__dirname, "../../src/asFunction.js"), babelConfig);
  expect(code).toMatchSnapshot();
});

it("MapDispatchToProps as arrowFuction", () => {
  const { code } = babel.transformFileSync(path.resolve(__dirname, "../../src/asArrowFunction.js"), babelConfig);
  expect(code).toMatchSnapshot();
});



