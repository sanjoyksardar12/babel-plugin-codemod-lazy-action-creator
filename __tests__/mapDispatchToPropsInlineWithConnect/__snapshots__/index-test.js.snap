// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDispatchToProps as arrowFuction 1`] = `
"import React, { Component } from \\"react\\";
import { connect } from \\"react-redux\\";
import { additionFour } from \\"./actions/addition.js\\";

class App extends Component {
  componentDidMount() {
    additionFour(2, 3, 4, 5);
    setTimeout(() => {
      const {
        multiplication2,
        multiplication3
      } = this.props;
      multiplication2(2, 3);
      setTimeout(() => {
        multiplication3(2, 3, 4);
      }, 1000);
    }, 1000 * 2);
    setTimeout(() => {
      const {
        addition2,
        addition3
      } = this.props;
      addition2(2, 3, 4);
      setTimeout(() => {
        addition3(2, 3, 4);
      }, 1000);
    }, 1000 * 4);
    setTimeout(() => {
      const {
        subtraction2,
        subtraction3
      } = this.props;
      subtraction2(2, 3);
      setTimeout(() => {
        subtraction3(2, 3, 4);
      }, 1000);
    }, 1000 * 6);
  }

  render() {
    const {
      addition,
      subtraction,
      multiplication
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"App\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Map dispatchToProps as function\\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Multiplication: \\", multiplication, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Addition: \\", addition, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Subtraction: \\", subtraction, \\" \\"));
  }

}

const mapStateToProps = state => {
  return { ...state
  };
};

const mapDispatchToProps = dispatch => {
  return {
    multiplication2: (...rest) => import(\\"./actions/multiplication.js\\").then(({
      default: multiplication
    }) => {
      dispatch(multiplication(...rest));
    }),
    multiplication3: (...rest) => import(\\"./actions/multiplication.js\\").then(multiplicationActions => {
      dispatch(multiplicationActions.multiplication3(...rest));
    }),
    addition2: (...rest) => import(\\"./actions/addition.js\\").then(({
      default: addition
    }) => {
      dispatch(addition(...rest));
    }),
    addition3: (...rest) => import(\\"./actions/addition.js\\").then(({
      additionThree: additionThree
    }) => {
      dispatch(additionThree(...rest));
    }),
    subtraction2: (...rest) => import(\\"./actions/subtraction.js\\").then(({
      default: subtraction
    }) => {
      dispatch(subtraction(...rest));
    }),
    subtraction3: (...rest) => import(\\"./actions/subtraction.js\\").then(({
      subtractionThree: subThree
    }) => {
      dispatch(subThree(...rest));
    })
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);"
`;

exports[`MapDispatchToProps as fuction 1`] = `
"import React, { Component } from \\"react\\";
import { connect } from \\"react-redux\\";
import { additionFour } from \\"./actions/addition.js\\";

class App extends Component {
  componentDidMount() {
    additionFour(2, 3, 4, 5);
    setTimeout(() => {
      const {
        multiplication2,
        multiplication3
      } = this.props;
      multiplication2(2, 3);
      setTimeout(() => {
        multiplication3(2, 3, 4);
      }, 1000);
    }, 1000 * 2);
    setTimeout(() => {
      const {
        addition2,
        addition3
      } = this.props;
      addition2(2, 3, 4);
      setTimeout(() => {
        addition3(2, 3, 4);
      }, 1000);
    }, 1000 * 4);
    setTimeout(() => {
      const {
        subtraction2,
        subtraction3
      } = this.props;
      subtraction2(2, 3);
      setTimeout(() => {
        subtraction3(2, 3, 4);
      }, 1000);
    }, 1000 * 6);
  }

  render() {
    const {
      addition,
      subtraction,
      multiplication
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"App\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Map dispatchToProps as function\\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Multiplication: \\", multiplication, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Addition: \\", addition, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Subtraction: \\", subtraction, \\" \\"));
  }

}

const mapStateToProps = state => {
  return { ...state
  };
};

function mapDispatchToProps(dispatch) {
  return {
    multiplication2: (...rest) => import(\\"./actions/multiplication.js\\").then(({
      default: multiplication
    }) => {
      dispatch(multiplication(...rest));
    }),
    multiplication3: (...rest) => import(\\"./actions/multiplication.js\\").then(multiplicationActions => {
      dispatch(multiplicationActions.multiplication3(...rest));
    }),
    addition2: (...rest) => import(\\"./actions/addition.js\\").then(({
      default: addition
    }) => {
      dispatch(addition(...rest));
    }),
    addition3: (...rest) => import(\\"./actions/addition.js\\").then(({
      additionThree: additionThree
    }) => {
      dispatch(additionThree(...rest));
    }),
    subtraction2: (...rest) => import(\\"./actions/subtraction.js\\").then(({
      default: subtraction
    }) => {
      dispatch(subtraction(...rest));
    }),
    subtraction3: (...rest) => import(\\"./actions/subtraction.js\\").then(({
      subtractionThree: subThree
    }) => {
      dispatch(subThree(...rest));
    })
  };
}

;
export default connect(mapStateToProps, mapDispatchToProps)(App);"
`;

exports[`MapDispatchToProps as inline with connnect 1`] = `
"import React, { Component } from \\"react\\";
import { connect } from \\"react-redux\\";
import { additionFour } from \\"./actions/addition.js\\";

class App extends Component {
  componentDidMount() {
    additionFour(2, 3, 4, 5);
    setTimeout(() => {
      const {
        multiplication2,
        multiplication3
      } = this.props;
      multiplication2(2, 3);
      setTimeout(() => {
        multiplication3(2, 3, 4);
      }, 1000);
    }, 1000 * 2);
    setTimeout(() => {
      const {
        addition2,
        addition3
      } = this.props;
      addition2(2, 3, 4);
      setTimeout(() => {
        addition3(2, 3, 4);
      }, 1000);
    }, 1000 * 4);
    setTimeout(() => {
      const {
        subtraction2,
        subtraction3
      } = this.props;
      subtraction2(2, 3);
      setTimeout(() => {
        subtraction3(2, 3, 4);
      }, 1000);
    }, 1000 * 6);
  }

  render() {
    const {
      addition,
      subtraction,
      multiplication
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"App\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Map dispatchToProps as inline with connect\\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Multiplication: \\", multiplication, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Addition: \\", addition, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Subtraction: \\", subtraction, \\" \\"));
  }

}

const mapStateToProps = state => {
  return { ...state
  };
};

export default connect(mapStateToProps, dispatch => ({
  multiplication2: (...rest) => import(\\"./actions/multiplication.js\\").then(({
    default: multiplication
  }) => dispatch(multiplication(...rest))),
  addition2: (...rest) => import(\\"./actions/addition.js\\").then(({
    default: addition
  }) => dispatch(addition(...rest))),
  addition3: (...rest) => import(\\"./actions/addition.js\\").then(({
    additionThree: additionThree
  }) => dispatch(additionThree(...rest))),
  subtraction2: (...rest) => import(\\"./actions/subtraction.js\\").then(({
    default: subtraction
  }) => dispatch(subtraction(...rest))),
  subtraction3: (...rest) => import(\\"./actions/subtraction.js\\").then(({
    subtractionThree: subThree
  }) => dispatch(subThree(...rest))),
  multiplication3: (...rest) => import(\\"./actions/multiplication.js\\").then(({
    multiplication3: multiplication3
  }) => dispatch(multiplication3(...rest)))
}))(App);"
`;

exports[`MapDispatchToProps as object 1`] = `
"import React, { Component } from \\"react\\";
import { connect } from \\"react-redux\\";
import { additionFour } from \\"./actions/addition.js\\";

class App extends Component {
  componentDidMount() {
    additionFour(2, 3, 4, 5);
    setTimeout(() => {
      const {
        multiplication2,
        multiplication3
      } = this.props;
      multiplication2(2, 3);
      setTimeout(() => {
        multiplication3(2, 3, 4);
      }, 1000);
    }, 1000 * 2);
    setTimeout(() => {
      const {
        addition2,
        addition3
      } = this.props;
      addition2(2, 3, 4);
      setTimeout(() => {
        addition3(2, 3, 4);
      }, 1000);
    }, 1000 * 4);
    setTimeout(() => {
      const {
        subtraction2,
        subtraction3
      } = this.props;
      subtraction2(2, 3);
      setTimeout(() => {
        subtraction3(2, 3, 4);
      }, 1000);
    }, 1000 * 6);
  }

  render() {
    const {
      addition,
      subtraction,
      multiplication
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"App\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Map dispatchToProps as object\\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Multiplication: \\", multiplication, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Addition: \\", addition, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Subtraction: \\", subtraction, \\" \\"));
  }

}

const mapStateToProps = state => {
  return { ...state
  };
};

const mapdispatchToProps = dispatch => {
  return {
    multiplication2: (...rest) => import(\\"./actions/multiplication.js\\").then(({
      default: multiplication
    }) => dispatch(multiplication(...rest))),
    addition2: (...rest) => import(\\"./actions/addition.js\\").then(({
      default: addition
    }) => dispatch(addition(...rest))),
    addition3: (...rest) => import(\\"./actions/addition.js\\").then(({
      additionThree: additionThree
    }) => dispatch(additionThree(...rest))),
    subtraction2: (...rest) => import(\\"./actions/subtraction.js\\").then(({
      default: subtraction
    }) => dispatch(subtraction(...rest))),
    subtraction3: (...rest) => import(\\"./actions/subtraction.js\\").then(({
      subtractionThree: subThree
    }) => dispatch(subThree(...rest))),
    multiplication3: (...rest) => import(\\"./actions/multiplication.js\\").then(({
      multiplication3: multiplication3
    }) => dispatch(multiplication3(...rest)))
  };
};

export default connect(mapStateToProps, mapdispatchToProps)(App);"
`;
